src/event_loop.cpp

```diff
@@ -87,6 +87,11 @@ int pollForEvents(void*) {
   // can be throttled)
   s_pollingRequested = 0;
 
+  // Periodically flush stdout/stderr buffers to ensure that any output from
+  // long-running Python calls is visible in the R console.
+  if (flush_std_buffers() != 0)
```

Comment:
If this fails, I think we should either:

1. Stop attempting to flush buffers;
2. Only log the warning once.

My main worry is that this background thread might fail to flush buffers repeatedly and so overload the console (if something got messed up with Python's stdout / stderr streams).